packages/cli/src/utilities.ts

```diff
@@ -0,0 +1,70 @@
+import { execSync } from 'child_process';
+import { createWriteStream } from 'fs';
+
+import chalk from 'chalk';
+import got, { Progress } from 'got';
+import { HttpsProxyAgent } from 'hpagent';
+import ora from 'ora';
+
+export const safeExecSync = (command: string) => {
+  try {
+    return execSync(command, { encoding: 'utf8', stdio: 'pipe' });
+  } catch {}
+};
+
+type Log = Readonly<{
+  info: typeof console.log;
+  warn: typeof console.log;
+  error: typeof console.log;
+}>;
+
+export const log: Log = Object.freeze({
+  info: (...args) => {
+    console.log(chalk.blue('[info]'), ...args);
+  },
+  warn: (...args) => {
+    console.log(chalk.yellow('[warn]'), ...args);
+  },
+  error: (...args) => {
+    console.log(chalk.red('[error]'), ...args);
+    // eslint-disable-next-line unicorn/no-process-exit
+    process.exit(1);
```

Comment:
nit: prefer to keep it at the outer scope, so the logging behavior is clean.

Reply:
in most cases it should end immediately. maybe we can provide another function like `log.message('error', ...args)` to achieve that?